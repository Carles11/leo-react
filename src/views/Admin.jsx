import React from "react";
import { Redirect } from "react-router-dom";

import API from "../utils/API"
import SignIn from "../components/SignIn";
import withAuth from '../components/withAuth'
import config from '../config'

class Admin extends React.Component {
  state = {
    message: '',
    next: false
  }

  handleData = async data => {
    const promise = await API.post('signin', data, true)
    const result = await promise.json()

    if (result.success) {
      localStorage.setItem(config.api.API_TOKEN, result.data)
      this.props.checkAuth()
    } else {
      console.log(result)
      this.setState({ message: result.data, next: true})
    }
  };

  render() {
    const { message, next } = this.state
    const { auth } = this.props;

    return (
      <React.Fragment>
        {!auth && <SignIn handleSubmit={this.handleData} message={message} next={next} />}
        {auth && <Redirect to="/admin-panel" />}
      </React.Fragment>
    );
  }
}

const AdminWithAuth = withAuth(Admin)

export default AdminWithAuth;
