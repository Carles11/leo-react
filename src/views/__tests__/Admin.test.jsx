import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import Admin from '../Admin'

const properties = {
  auth: false,
  checkAuth: jest.fn(),
}

function setup() {
  const props = properties

  const enzymeWrapper = mount(
    <MemoryRouter initialEntries={['/test']}>
      <Admin {...props} />
    </MemoryRouter>,
  )

  return {
    props,
    enzymeWrapper,
  }
}

describe('<Admin />', () => {
  it('should render itself', () => {
    const component = shallow(<Admin {...properties} />)
    expect(component).toMatchSnapshot()
  })

  it('should render SignIn Component if auth is false', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.find('SignIn').length).toEqual(1)
  })

  it('should call checkAuth prop method', () => {
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.find('SignIn').length).toEqual(1)
  })

  it('should redirect to admin-panel if auth is true', () => {
    properties.auth = true
    const { enzymeWrapper } = setup()
    expect(enzymeWrapper.find('SignIn').length).toEqual(0)
    expect(enzymeWrapper.find('Redirect').length).toEqual(1)
  })
})
