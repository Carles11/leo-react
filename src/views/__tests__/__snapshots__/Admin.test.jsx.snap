// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin /> should render itself 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Admin
    auth={false}
    checkAuth={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Component
          handleSubmit={[Function]}
          message=""
          next={false}
/>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSubmit": [Function],
          "message": "",
          "next": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Component
            handleSubmit={[Function]}
            message=""
            next={false}
/>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSubmit": [Function],
            "message": "",
            "next": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
